gocourse16/main.go

```diff
@@ -0,0 +1,81 @@
+/*
+Виконати пошук телефонних номерів у файлі з даними контактів. Задача полягає в тому, щоб створити регулярний вираз,
+який можна використовувати для знаходження телефонних номерів, записаних у різних форматах. Наприклад, ви можете почати
+з використання виразу, який знаходить номери телефонів, що складаються з 10 цифр, а потім розширити його, додавши
+підтримку різних форматів, наприклад, номери з круглими дужками, пробілами та дефісами. Для відпрацювання регулярного
+виразу дивіться https://regex101.com/. Далі треба перезаписати цей файл із іменами та номерами телефонів, щоб усі
+номери були в одному форматі, а саме (000) 11-22-333.
+*/
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+)
+
+var regexPhone = regexp.MustCompile(`[^\d]`)
+
+func TrimNumber(s string) string {
+	cleanNumber := regexPhone.ReplaceAllString(s, "")
+	if strings.HasPrefix(cleanNumber, "0") {
+		cleanNumber = "380" + cleanNumber[1:]
+	} else if strings.HasPrefix(cleanNumber, "+380") {
+		cleanNumber = strings.TrimPrefix(cleanNumber, "+")
+	}
+	if len(cleanNumber) != 12 {
+		return "Invalid phone number"
```

Comment:
В такому випадку має повертати помилку `error` та порожній рядок. Так потім легше обробляти помилки та код виходить консистентним з іншим Go кодом:
```suggestion
		return "", errors.New("invalid phone number")
```